### PR TITLE
Feature/redux recompose

### PR DESCRIPTION
**Pull request by ghost**
## Summary

* Install and configure redux-recompose
* Refactor actions and reducers to work with redux-recompose
* Apply fetchMiddleware

## Trello Card

[Trello Card](https://trello.com/c/0kR0PKbd/23-n-97-redux-recompose-ii-refactor-book-reducer)

### CODE REVIEW

File: wbooks/src/redux/book/actions.ts [hunk 1]
```diff
@@ -1,22 +1,16 @@
...
 import { getBooks } from '@services/BookService';
 
-export const actions = {
-  GET_BOOKS: '@@BOOK/GET_BOOKS',
-  GET_BOOKS_SUCCESS: '@@BOOK/GET_BOOKS_SUCCESS',
-  GET_BOOKS_FAILURE: '@@BOOK/GET_BOOKS_FAILURE'
-};
+export const actions = createTypes(completeTypes({ primaryActions: ['GET_BOOKS'] }), '@@BOOK');
+
+const target = 'books';
```
Comment (sfernandez11) [inline wbooks/src/redux/book/actions.ts]: constants must be in UPPERCASE

File: wbooks/src/redux/book/reducer.ts [hunk 2]
```diff
@@ -1,29 +1,16 @@
...
-      return { ...state, loading: true };
-    case actions.GET_BOOKS_SUCCESS:
-      return { ...state, books: action.payload as Book[], loading: false };
-    case actions.GET_BOOKS_FAILURE:
-      return { ...state, loading: false, error: action.payload as string };
-    default:
-      return state;
-  }
-}
+const initialState: BookState = completeState({ description: bookState });
```
Comment (sfernandez11) [inline wbooks/src/redux/book/reducer.ts]: Fix this! I think this must be:
```
const stateDescription = {
  description: {
    books: []
  }
};

export const initialState: : BookState = completeState(stateDescription);
```

**Review by sfernandez11 — CHANGES_REQUESTED** [review]

File: wbooks/src/redux/book/actions.ts (hunk 1 above)
Comment (ghost) [reply wbooks/src/redux/book/actions.ts]: Done!

File: wbooks/src/redux/book/reducer.ts (hunk 2 above)
Comment (ghost) [reply wbooks/src/redux/book/reducer.ts]: Why we need to export `initialState`?

**Review by sfernandez11 — APPROVED** [review]

**Pull request merged**